File: 백준/Bronze/2581. 소수/소수.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        boolean[] checked = new boolean[10000 + 1];
        checked[0] = checked[1] = true;

        for (int i = 2; i * i <= checked.length - 1; i++) {
            if (checked[i]) {
                continue;
            }

            for (int j = i * i; j <= checked.length - 1; j += i) {
                checked[j] = true;
            }
        }

        int n = sc.nextInt();
        int m = sc.nextInt();

        int sum = 0;
        int min = Integer.MAX_VALUE;

        for (int i = n; i <= m; i++) {
            if (!checked[i]) {
                min = Math.min(i, min);
                sum += i;
            }
        }

        if (sum != 0) {
            System.out.println(sum);
            System.out.println(min);
        } else {
            System.out.println(-1);
        }
    }
}

File: 백준/Bronze/2581. 소수/소수.py
import sys

# sys.stdin = open("input.txt", "r")

checked = [False] * 10001
checked[0] = checked[1] = True

for i in range(2, 10001):
    if checked[i] == True:
        continue

    for j in range(i * i, 10001, i):
        checked[j] = True

n = int(sys.stdin.readline().strip())
m = int(sys.stdin.readline().strip())

sum = 0
min_num = 10000

for i in range(n, m + 1):
    if checked[i] == False:
        sum += i
        min_num = min(min_num, i)

if sum == 0:
    print(-1)
else:
    print(sum)
    print(min_num)